html/location.html

```diff
@@ -0,0 +1,67 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+    <title>Cacophonator Setup</title>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="stylesheet" href="/static/css/bootstrap.min.css">
+</head>
+
+<body>
+    {{template "navbar"}}
+
+    <div class="container">
+        <div 
+            id="resultMessage" class="alert alert-danger" 
+            {{if .ErrorEncountered}} 
+                style="display: block;" 
+            {{else}} 
+                style="display: none;" 
+            {{end}}> {{.ErrorMessage}}
+        </div>
+
+        <div class="container pt-5 pl-0">
+            <h2>Device Location<br></h2>
+        </div>
+        <hr>
+
+
+        <form action="/location" method="POST"> 
+
+            <div class="row">
+                <div class="form-group col-md-4"> 
+                    <label for="latitude">Latitude</label>
+                    <input type="text" class="form-control" id="lat" name="latitude"
+                    {{if eq .Location.Latitude 0.0}} 
+                        placeholder="Enter latitude"
+                    {{else}}
+                        value="{{.Location.Latitude}}"
+                    {{end}}>
+                </div>
+            </div>
+
+            <div class="row">
+                <div class="form-group col-md-4">
+                    <label for="longitude">Longitude</label>
+                    <input type="text" class="form-control" id="long" name="longitude"
+                    {{if eq .Location.Longitude 0.0}} 
+                        placeholder="Enter longitude"
+                    {{else}}
+                        value="{{.Location.Longitude}}"
+                    {{end}}>
+                </div>
+            </div>
+
+        <button type="submit" class="btn btn-primary">Submit</button> 
```

Comment:
To avoid the weird URL issue, don't have the form post to `/api/location` directly but instead attach some JavaScript to the submit button which uses jQuery to do the post and if that succeeds, reload the page.

See https://api.jquery.com/jQuery.post/. There's a full example of doing most of this at the bottom.

This avoids the URL changing after the submit butted is pressed and separates the UI part from the API part.

Reply:
Yeah, sigh, I did see code like this.  Seems to very complicated to do a very simple thing.

I find the syntax of JQuery and JS very confusing and just plain horrible to look at and try and figure out what is going on.

I'll get it done though.

Reply:
So I'm guessing I don't call the handler I have set up her right? router.HandleFunc("/api", managementinterface.LocationHandler).Methods("POST")

I call another Go routine which does similar stuff and returns success or fail right?